src/libs/actions/Welcome.js

```diff
@@ -124,6 +142,9 @@ function show({routes, showCreateMenu}) {
         if (!Policy.isAdminOfFreePolicy(allPolicies) && !isDisplayingWorkspaceRoute) {
             showCreateMenu();
         }
+
+        // Update state of isFirstTimeNewExpensifyUser so the Welcome logic doesn't run again
+        isFirstTimeNewExpensifyUser = false;
```

Comment:
Shouldn't we set this in Onyx instead?

Reply:
I couldn't make the feature work, specifically for mWeb/iOS/Android, if Onyx updated that key.

I discussed this further [here](https://github.com/Expensify/App/pull/16962#issuecomment-1507685866) and the solution came from this [slack thread](https://expensify.slack.com/archives/C03TQ48KC/p1681417598780039).

A brief explanation is:
In small screen devices, because of how the LNH works, some components are mounted twice which causes 3 calls to `Welcome.show` to be made.

Only the third call to `Welcome.show` can open the Popover menu. The problem is that by the time the 3rd `Welcome.show` call runs the device already synced up with Onyx and updated the state of that key to false : /

This is the best solution I could come up with so that key is still `true` by the time `Welcome.show` runs.

Reply:
Not sure if that was the question @luacmartins so please lmk!

Reply:
Ah makes sense! Thanks!